rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeperClient.java

```diff
@@ -53,14 +55,16 @@ public void close()
     }
 
     // Caller responsible to close the client
-    public static BookKeeperClient createBookKeeperClient(Configuration conf)
+    public static RetryingBookkeeperClient createBookKeeperClient(Configuration conf)
             throws TTransportException
     {
         TTransport transport;
-        transport = new TSocket("localhost", CacheConfig.getServerPort(conf));
+        transport = new TSocket("localhost", CacheConfig.getServerPort(conf), CacheConfig.getClientTimeout(conf));
+
         transport.open();
 
         BookKeeperClient client = new BookKeeperClient(transport);
-        return client;
+        RetryingBookkeeperClient retryingBookkeeperClient = new RetryingBookkeeperClient(client, CacheConfig.getMaxRetries(conf));
```

Comment:
Forgot to catch this before, but creating the BookKeeperClient seems redundant. We can very well move the logic of BookKeeperClient.getCacheStatus etc into RetryinBookKeeperClient.

Do we need to keep BookKeeperClient around for any other use? If not lets get rid of it and only keep RetryinBookKeeperClient.